File: backend/src/main/java/com/example/tbmini/domain/entity/Board.java
package com.example.tbmini.domain.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 看板（板块）实体
 * 对应数据库中的boards表
 */
@Data
@TableName("boards")
public class Board {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String slug;

    private String description;

    private Boolean isActive;

    private Integer sortOrder;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: frontend/src/api/board.ts
import request from './request'

export interface Board {
  id: number
  name: string
  slug: string
  description: string
  createdAt: string
}

export interface BoardDetailResponse {
  id: number
  name: string
  slug: string
  description: string
  isActive: boolean
  createdAt: string
}

export function listBoards() {
  return request.get<Board[]>('/boards')
}

export function getBoardBySlug(slug: string) {
  return request.get<BoardDetailResponse>(`/boards/${slug}`)
}

export function listBoardPosts(slug: string, params: { page?: number; pageSize?: number }) {
  return request.get(`/boards/${slug}/posts`, { params })
}
